daffodil-core/src/main/scala/org/apache/daffodil/dsom/SchemaSet.scala

```diff
@@ -62,22 +63,25 @@ import scala.collection.immutable.HashMap
  */
 
 final class SchemaSet(
+  pfArg: => Option[ProcessorFactory],
   rootSpec: Option[RootSpec],
   externalVariables: Seq[Binding],
   schemaSourcesArg: Seq[DaffodilSchemaSource],
   val validateDFDLSchemas: Boolean,
   checkAllTopLevelArg: Boolean,
-  parent: SchemaComponent,
   tunableArg: DaffodilTunables)
-  extends SchemaComponentImpl(<schemaSet/>, parent) // a fake schema component
+  extends SchemaComponentImpl(<schemaSet/>, null)
   with SchemaSetIncludesAndImportsMixin {
 
+  private lazy val processorFactory = pfArg
+  lazy val root = rootElement(processorFactory.flatMap { _.rootSpec })
+
```

Comment:
What's the intention for the pfArg being passed in and the creation of the root variable? From what I can tell, root isn't used.

Reply:
root is used many places where optimizers ask about global properties of the schema's root element., e..g, schemaSet.root.isScannable tells us if the whole schema (for this root anyway) is all scannable text.